cmd/thanos/sidecar.go

```diff
@@ -371,6 +383,33 @@ func runSidecar(
 	return nil
 }
 
+func isPrometheusAgentModeEnabled(ctx context.Context, client *promclient.Client, logger log.Logger, m *promMetadata) (bool, error) {
```

Comment:
I wonder if we can refactor `validatePrometheus` method somehow to return this information?